src/main/java/org/spongepowered/common/mixin/core/item/MixinItemFishingRod.java

```diff
@@ -54,10 +54,12 @@ public ItemStack onItemRightClick(ItemStack itemStack, World world, EntityPlayer
             player.fishEntity.handleHookRetraction();
         } else {
             EntityFishHook fishHook = new EntityFishHook(world, player);
-            if (!Sponge.getGame().getEventManager()
+            if (world.isRemote || !Sponge.getGame().getEventManager()
```

Comment:
I would prefer to force plugin developers to not register things on the client instead of going down the per side method/class stripping while we still have no client api's

It is my opinion that plugins should be informing users that they are useless on a particular side instead of just causing the errors and crashes that would end up occurring if you added side attributes to @plugin or put side only annotations on your main plugin class.
> On Jul 27, 2015, at 11:57 AM, JBYoshi <notifications@github.com> wrote:
> 
> In src/main/java/org/spongepowered/common/mixin/core/item/MixinItemFishingRod.java:
> 
> > @@ -54,10 +54,12 @@ public ItemStack onItemRightClick(ItemStack itemStack, World world, EntityPlayer
> >              player.fishEntity.handleHookRetraction();
> >          } else {
> >              EntityFishHook fishHook = new EntityFishHook(world, player);
> > -            if (!Sponge.getGame().getEventManager()
> > +            if (world.isRemote || !Sponge.getGame().getEventManager()
> @Zidane
> 
> Introduction of another enum is un-necessary. The one available in Platform would suffice.
> 
> How do you handle BOTH or DEFAULT? Values in annotations can't be set to null.
> 
> Scratch bullet 3, don't want to go down the route of "Side Only classes" at this time.
> 
> Sounds good. If someone wants to do something like that, either something on the @Plugin annotation would suffice, or they can register the listener only on the client.
> 
> if (game.getPlatform().getType() == Platform.Type.CLIENT) { 
>     game.getEventManager().register(this, new ClientSideListener());
> }
> —
> Reply to this email directly or view it on GitHub.
>